test/unit/erase.js

```diff
@@ -0,0 +1,82 @@
+var sinon = require('sinon');
+var Tessel = require('../../lib/tessel/tessel');
+var commands = require('../../lib/tessel/commands');
+var logs = require('../../lib/logs');
+var TesselSimulator = require('../common/tessel-simulator');
+
+exports['Tessel.prototype.erase'] = {
+  setUp: function(done) {
+
+    this.erase = sinon.spy(Tessel.prototype, 'eraseScript');
+    this.logsWarn = sinon.stub(logs, 'warn', function() {});
+    this.logsInfo = sinon.stub(logs, 'info', function() {});
+
+    this.tessel = TesselSimulator();
+
+    done();
+  },
+
+  tearDown: function(done) {
+    this.tessel.close();
+    this.erase.restore();
+    this.logsWarn.restore();
+    this.logsInfo.restore();
+    done();
+  },
+
+  eraseAsUsual: function(test) {
+    var self = this;
+
+    test.expect(3);
+
+    // Test that we received the proper command
+    self.tessel._rps.once('command', function(command) {
+      var receivedCommands = command.toString().split(' ');
+      var expected = commands.stopRunningScript();
+      // Test that the command has the proper number of args
+      test.equal(receivedCommands.length, expected.length);
+
+      // And the proper args in each place
+      for (var i = 0; i < receivedCommands.length; i++) {
+        test.equal(receivedCommands[i], expected[i]);
+      }
+
+    });
+
+    self.tessel.eraseScript()
+      .then(function() {
+        // This test completed satisfactorily
+        test.done();
+      })
+      .catch(function() {
+        test.fail('Error thrown on proper flash erase');
+      });
+
+    self.tessel._rps.emit('close');
+  },
+
+  noCodeInFlash: function(test) {
```

Comment:
Add: 

```
test.expect(1);
```

Otherwise, if `test.done()` is reached but no assertions have been executed nodeunit will consider that a "pass".